src/Apps/Order/Routes/__tests__/Payment.test.tsx

```diff
@@ -241,4 +249,65 @@ describe("Payment", () => {
 
     expect(testProps.router.push).toHaveBeenCalledWith("/order2/1234/review")
   })
+
+  it("shows an error modal when there is an error in CreateCreditCardPayload", () => {
+    stripeMock.createToken.mockReturnValue({
+      then: func => func({ token: { id: "tokenId" } }),
+    })
+
+    mutationMock.mockImplementationOnce((_, { onCompleted }) =>
+      onCompleted(creatingCreditCardFailed)
+    )
+
+    const component = mount(<PaymentRoute {...testProps} />)
+
+    expect(component.find(ErrorModal).props().show).toBe(false)
+
+    component.find(ContinueButton).simulate("click")
+
+    expect(component.find(ErrorModal).props().show).toBe(true)
+    expect(component.find(ErrorModal).props().detailText).toBe(
+      "No such token: fake-token"
```

Comment:
Not sure if this is helpful enough from a user's perspective, but at least this will add a foundation that shows a message from the server and we should be able to change it easily.